crypto/fipsmodule/cpucap/cpu_aarch64.c

```diff
@@ -49,4 +49,44 @@ void handle_cpu_env(uint32_t *out, const char *in) {
   }
 }
 
+#if defined(MAKE_DIT_AVAILABLE) && !defined(OPENSSL_WINDOWS)
+// "DIT" is not recognised as a register name by clang-10 (at least)
+// Register's encoded name is from e.g.
+// https://github.com/ashwio/arm64-sysreg-lib/blob/d421e249a026f6f14653cb6f9c4edd8c5d898595/include/sysreg/dit.h#L286
+#define DIT_REGISTER s3_3_c4_c2_5
+
+uint64_t armv8_get_dit(void) {
+  uint64_t val = 0;
+  if (CRYPTO_is_ARMv8_DIT_capable()) {
+    __asm__ volatile("mrs %0, s3_3_c4_c2_5" : "=r" (val));
+  }
+  return val >> 24;
```

Comment:
if the intention is to extract bit 24 only then we should do: `(val >> 24) & 1`?

Reply:
Done, thanks. I had missed that because in the printout tests, it printed as 1.